Add instructions to install a PTF on an air-gapped server (#3904)
uyuni-project/uyuni-docs: modules/installation-and-upgrade/pages/container-deployment/mlm/server-air-gapped-deployment-mlm.adoc

For more detailed information about installing {productname} Server on {sl-micro}, see xref:container-deployment/mlm/server-deployment-mlm.adoc[Deploy Server as a Virtual Machine].

To upgrade {productname} Server, users should upgrade all packages in the system and follow the procedures defined in xref:container-management/updating-server-containers.adoc[Server Upgrade].

== PTFs

The PTF images are not available as packages.
This means that they should be pulled using [command]``podman`` on a machine with internet access, then saved in an archive, transferred to the air-gapped machine and loaded there.

.Procedure: Pull the image on a machine with internet access
. Install [command]``podman``
. Authenticate against the SUSE Registry using the SCC credentials:
+
[source,shell]
----
set +o history
echo SCC_MIRRORING_PASSWORD | podman login -u "SCC_MIRRORING_USER" --password-stdin registry.suse.com
set -o history
----

> - Create a [path]``/tmp/ptf-images`` temporary file with the URL of the PTF images.
> - Pull each of the container images of the PTF and save them in a tar archive
> +

Reviewer: ```suggestion
. Create a [path]``/tmp/ptf-images`` temporary file with the URL of the PTF images.
. Pull each of the container images of the PTF and save them in a tar archive.

+

```

[source,shell]
----
for image in `cat /tmp/ptf-images`; do
    podman pull $image
done
podman save -o /tmp/ptf-images.tar `cat /tmp/ptf-images`
----

> - Transfer the [path]``/tmp/ptf-images.tar`` images archive on the server to patch.

Reviewer: ```suggestion
. Transfer the [path]``/tmp/ptf-images.tar`` images archive on the server to patch.
```

> .Procedure: Load the images on the server to patch
> . Ensure the [path]``ptf-images.tar`` file is available on the server.
> . Load the images from the archive
> +

Reviewer: ```suggestion
.Procedure: Loading the images on the server to patch
. Ensure the [path]``ptf-images.tar`` file is available on the server.
. Load the images from the archive:

+

```

[source,shell]
----
podman load -i ptf-images.tar
----

> - Install the PTF using the `--pullPolicy=never` parameter since the images are already available.

Reviewer: ```suggestion
. Install the PTF using the [option]``--pullPolicy=never`` parameter because the images are already available.
```

For more detailed information about installing {productname} Server on {sl-micro}, see xref:container-deployment/mlm/server-deployment-mlm.adoc[Deploy Server as a Virtual Machine].

To upgrade {productname} Server, users should upgrade all packages in the system and follow the procedures defined in xref:container-management/updating-server-containers.adoc[Server Upgrade].

== PTFs

The PTF images are not available as packages.
This means that they should be pulled using [command]``podman`` on a machine with internet access, then saved in an archive, transferred to the air-gapped machine and loaded there.

.Procedure: Pulling the image on a machine with internet access
. Install [command]``podman``.
. Authenticate against the {suse} Registry using the SCC credentials:

+

[source,shell]
----
set +o history
echo SCC_MIRRORING_PASSWORD | podman login -u "SCC_MIRRORING_USER" --password-stdin registry.suse.com
set -o history
----
. Create a [path]``/tmp/ptf-images`` temporary file with the URL of the PTF images, one per line.

> In most of the cases only the server image is needed and it can be created with a command like the following, after replacing the [command]``SCC_USERID`` and [command]``PTFID`` values.

Reviewer: ```suggestion
  In most of the cases only the server image is needed and it can be created with a command like the following, after replacing the [command]``SCC_USERID`` and [command]``PTFID`` values.
```

+
[source,shell]
----
SCC_USERID=aXXXX
PTFID=12345
echo "registry.suse.com/a/$SCC_USERID/$PTFID/multi-linux-manager/5.1/x86_64/server:latest-ptf-$PTFID" >>/tmp/ptf-images
----

. Pull each of the container images of the PTF and save them in a tar archive.

+

[source,shell]
----
for image in `cat /tmp/ptf-images`; do
    podman pull $image
done
podman save -o /tmp/ptf-images.tar `cat /tmp/ptf-images`
----
. Transfer the [path]``/tmp/ptf-images.tar`` images archive on the server to patch.

.Procedure: Loading the images on the server to patch
. Ensure the [path]``ptf-images.tar`` file is available on the server.
. Load the images from the archive:

+

[source,shell]
----
podman load -i ptf-images.tar
----
. Install the PTF using `mgradm support ptf podman` as would be done on a connected machine.

> Because the images are already loaded they won't be pulled.

Reviewer: ```suggestion
  Because the images are already loaded they will not be pulled.
```
